src/rabbit_vhost.erl

```diff
@@ -61,37 +65,47 @@ recover(VHost) ->
     ok = rabbit_mirror_queue_misc:on_vhost_up(VHost),
     ok.
 
-%%----------------------------------------------------------------------------
-
--define(INFO_KEYS, [name, tracing, cluster_state]).
+-define(INFO_KEYS, vhost:info_keys()).
 
--spec add(rabbit_types:vhost(), rabbit_types:username()) -> rabbit_types:ok_or_error(any()).
+-spec add(vhost:name(), rabbit_types:username()) -> rabbit_types:ok_or_error(any()).
 
 add(VHost, ActingUser) ->
     case exists(VHost) of
         true  -> ok;
-        false -> do_add(VHost, ActingUser)
+        false -> do_add(VHost, <<"">>, <<"">>, ActingUser)
+    end.
+
+-spec add(vhost:name(), rabbit_types:username(), string(), [atom()]) -> rabbit_types:ok_or_error(any()).
+
+add(Name, Description, Tags, ActingUser) ->
+    case exists(Name) of
+        true  -> ok;
+        false -> do_add(Name, Description, Tags, ActingUser)
     end.
 
-do_add(VHostPath, ActingUser) ->
-    rabbit_log:info("Adding vhost '~s'~n", [VHostPath]),
-    R = rabbit_misc:execute_mnesia_transaction(
+do_add(Name, Description, Tags, ActingUser) ->
+    rabbit_log:info("Adding vhost '~s' (description: '~s', tags: '~s')", [Name, Description, Tags]),
```

Comment:
The type for tags in the `-spec()` doesn't match the type expected by this format string (list of atoms vs. a string), though I suppose `io:format()` will take this as an `iolist()` and silently concatenate all atoms.

Reply:
This error is still present. Furthermore, the CLI now passes a list of tags.